File: src/ProjectR.Scripting/content/scripts/spells/lua/Slash.lua
function GetTargetType()
    return Single
end

function GetName()
    return "Slash"
end

function GetDescription()
    return "A sword slash that ignores some of the targets DEF."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 0
end

function GetDelay()
  return .35
end

function GetSpellType()
  return Physical
end

function GetMasteries()
  return {  }
end

function SpellEffect()
    local damage = 1.5 * aAD - .75 * dDEF
    d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/Slash.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting
{
    public class Slash : SpellScriptBase
    {
        public override string Name { get { return "Slash"; } }
        public override string Description { get { return "A sword slash that ignores some of the targets DEF."; } }

        public override TargetType TargetType { get { return TargetType.Single; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { }; } }
        public override SpellType SpellType { get { return SpellType.Physical; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 0; } }
        public override double Delay { get { return .35; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 1.5 * aAD - .75 * dDEF;
            DealDamage(damage);
        }
    }
}